### ISSUE TITLE
DOWNLOAD_ONLY option doesn't work properly

### ISSUE THREAD

**Issue by kgerheiser** [NOAA-EMC/hpc-stack#141]
**Describe the bug**
DOWNLOAD_ONLY doesn't work properly if not connected.

The git fetch needs to be removed from PIO

The other issue with PIO is that it uses those CMake modules internally and attempts to download them and fails if there is no connection.

The problem with ESMF is that it's deleted every time before it's built because ESMF does not support out-of-source builds.

**To Reproduce**
Enable DOWNLOAD_ONLY and download the packages, then disable and build

**Expected behavior**

The build should work without an internet connection

**Comment by edwardhartnett** [comment]
I suggest you build PIO with the autotools build. I try to keep the CMake build up to date, but my co-programmer on PIO generally handles the CMake build. The autotools build does not do a git fetch, it includes everything it needs.

**Comment by kgerheiser** [comment]
@edwardhartnett I've been trying out the autotools build and I can build, but `make check` doesn't pass. All tests except these ones pass:

```
0 expected NetCDF: Not a valid ID got Unknown Error: Unrecognized error code
2 expected NetCDF: Not a valid ID got Unknown Error: Unrecognized error code
0 Error 1111 in ../../../tests/cunit/test_pioc.c, line 549
1 expected NetCDF: Not a valid ID got Unknown Error: Unrecognized error code
3 expected NetCDF: Not a valid ID got Unknown Error: Unrecognized error code
1 Error 1111 in ../../../tests/cunit/test_pioc.c, line 549
2 Error 1111 in ../../../tests/cunit/test_pioc.c, line 549
3 Error 1111 in ../../../tests/cunit/test_pioc.c, line 549
1 Error 1111 in ../../../tests/cunit/test_pioc.c, line 643
Attempting to use an MPI routine after finalizing MPICH
2 Error 1111 in ../../../tests/cunit/test_pioc.c, line 643
Attempting to use an MPI routine after finalizing MPICH
3 Error 1111 in ../../../tests/cunit/test_pioc.c, line 643
Attempting to use an MPI routine after finalizing MPICH
0 Error 1111 in ../../../tests/cunit/test_pioc.c, line 643
Attempting to use an MPI routine after finalizing MPICH
```

Something about testing NetCDF classic format. I'm building without pnetcdf, could that be why?

```
export LDFLAGS="-L$NetCDF_ROOT/lib -lnetcdf -L$HDF5_ROOT/lib -lhdf5 -lhdf5_hl -L$ZLIB_ROOT -lz -ldl -lm"

./configure --enable-fortran --disable-pnetcdf
```

**Comment by kgerheiser** [comment]
The tests work when I use pnetcdf.

**Comment by edwardhartnett** [comment]
Yes, this is a bug in the PIO autotools build. However, we never encounter it because everyone who uses PIO also wants to sometimes (or always) use pnetcdf, so it's always enabled.

I will try to fix this next time I spend time on PIO, but that might not be for some time. I would suggest as a workaround that we go ahead and require pnetcdf for the PIO build.

**Comment by edwardhartnett** [comment]
Let's bump this issue to the 1.3.0 release of hpc-stack.

**Comment by kgerheiser** [comment]
Fixed with #360
